From 143ea7bc27559961b816cae4f9c4dbe407d02233 Mon Sep 17 00:00:00 2001
From: Herbert Xu <herbert@gondor.apana.org.au>
Date: Mon, 24 Sep 2007 21:59:39 +0800
Subject: [PATCH] [PARSER] Remove superfluous arinest test for dqvarnest

dqvarnest is only used to determine whether CENDQUOTE should terminate the
double-quote syntax.  Since CENDQUOTE can never occur while arinest is set,
we don't need to take arinest into account for dqvarnest.
---
 ChangeLog    |    1 +
 src/parser.c |    3 +--
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 0a35741..b311ad4 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -3,6 +3,7 @@
 	* Do not quote back slashes in parameter expansions outside quotes.
 	* Perform tilde expansion in all parameter expansion words.
 	* Remove superfluous arinest test in CENDQUOTE.
+	* Remove superfluous arinest test for dqvarnest.
 
 2007-09-22  Oleg Verych <olecom@flower.upol.cz>
 
diff --git a/src/parser.c b/src/parser.c
index 26c9046..8fac57a 100644
--- a/src/parser.c
+++ b/src/parser.c
@@ -1232,9 +1232,8 @@ badsub:			synerror("Bad substitution");
 		*((char *)stackblock() + typeloc) = subtype | flags;
 		if (subtype != VSNORMAL) {
 			varnest++;
-			if (dblquote || arinest) {
+			if (dblquote)
 				dqvarnest++;
-			}
 		}
 	}
 	goto parsesub_return;
-- 
1.5.3.7

